From 428b331d8efede1e2f39f2fc49216c675d081030 Mon Sep 17 00:00:00 2001
From: sehraf <sehraf42@gmail.com>
Date: Fri, 25 May 2018 23:12:35 +0200
Subject: [PATCH] fix for Qt 5.11

Quote from Arch mailing list:
- there's been a huge header cleanup in Qt modules. Expect build failures for applications that rely on transitive includes instead of declaring all required headers. Those need to be fixed upstream by explicitely adding the missing includes.
---
 retroshare-gui/src/gui/Posted/PostedItem.cpp        | 1 +
 retroshare-gui/src/gui/chat/ChatTabWidget.cpp       | 2 ++
 retroshare-gui/src/gui/feeds/GxsChannelPostItem.cpp | 1 +
 retroshare-gui/src/gui/feeds/GxsForumMsgItem.cpp    | 1 +
 4 files changed, 5 insertions(+)

diff --git a/retroshare-gui/src/gui/Posted/PostedItem.cpp b/retroshare-gui/src/gui/Posted/PostedItem.cpp
index 7d70b3e157..8fc1cc6869 100644
--- a/retroshare-gui/src/gui/Posted/PostedItem.cpp
+++ b/retroshare-gui/src/gui/Posted/PostedItem.cpp
@@ -22,6 +22,7 @@
  */
 
 #include <QDateTime>
+#include <QStyle>
 
 #include "rshare.h"
 #include "PostedItem.h"
diff --git a/retroshare-gui/src/gui/chat/ChatTabWidget.cpp b/retroshare-gui/src/gui/chat/ChatTabWidget.cpp
index a965bbb323..6dd9c27576 100644
--- a/retroshare-gui/src/gui/chat/ChatTabWidget.cpp
+++ b/retroshare-gui/src/gui/chat/ChatTabWidget.cpp
@@ -20,6 +20,8 @@
  *  Boston, MA  02110-1301, USA.
  ****************************************************************/
 
+#include <QTabBar>
+
 #include "ChatTabWidget.h"
 #include "ui_ChatTabWidget.h"
 #include "ChatDialog.h"
diff --git a/retroshare-gui/src/gui/feeds/GxsChannelPostItem.cpp b/retroshare-gui/src/gui/feeds/GxsChannelPostItem.cpp
index d154408b73..72317640c7 100644
--- a/retroshare-gui/src/gui/feeds/GxsChannelPostItem.cpp
+++ b/retroshare-gui/src/gui/feeds/GxsChannelPostItem.cpp
@@ -23,6 +23,7 @@
 
 #include <QTimer>
 #include <QFileInfo>
+#include <QStyle>
 
 #include "rshare.h"
 #include "GxsChannelPostItem.h"
diff --git a/retroshare-gui/src/gui/feeds/GxsForumMsgItem.cpp b/retroshare-gui/src/gui/feeds/GxsForumMsgItem.cpp
index 97ac9dd75a..e9773aba9c 100644
--- a/retroshare-gui/src/gui/feeds/GxsForumMsgItem.cpp
+++ b/retroshare-gui/src/gui/feeds/GxsForumMsgItem.cpp
@@ -23,6 +23,7 @@
 
 #include <QTimer>
 #include <QFileInfo>
+#include <QStyle>
 
 #include "rshare.h"
 #include "GxsForumMsgItem.h"
